pubsub-protocol-binding.md

```diff
@@ -0,0 +1,276 @@
+
+## Abstract
+
+The Google Cloud Pub/Sub Protocol Binding for CloudEvents defines how events are mapped to Pub/Sub
+1.1 request and response messages.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to Pub/Sub](#12-relation-to-pubsub)
+- 1.3. [Content Modes](#13-content-modes)
+- 1.4. [Event Formats](#14-event-formats)
+- 1.5. [Security](#15-security)
+
+2. [Use of CloudEvents Attributes](#2-use-of-cloudevents-attributes)
+
+- 2.1. [datacontenttype Attribute](#21-datacontenttype-attribute)
+- 2.2. [data Attribute](#22-data-attribute)
+
+3. [Pub/Sub Message Mapping](#3-pub/sub-message-mapping)
+
+- 3.1. [Binary Content Mode](#31-binary-content-mode)
+- 3.2. [Structured Content Mode](#32-structured-content-mode)
+
+4. [References](#4-references)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how
+the elements defined in the CloudEvents specification are to be used in 
+[Pub/Sub][pubsub] requests and response messages.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2. Relation to Pub/Sub
+
+Events can be produced using a Topic or consumed using Push or Pull based Subscriptions. 
+
+### 1.3. Content Modes
+
+The specification defines two content modes for transferring events:
+_structured_ and _binary_.
+
+In the _structured_ content mode, event metadata attributes and event data are
+placed into the Pub/Sub message body section using an
+[event format](#14-event-formats).
+
+In the _binary_ content mode, the value of the event `data` attribute is placed
+into the Pub/Sub payload data as-is, with the `datacontenttype`
+attribute value declaring its media type; all other event attributes are mapped
+to Pub/Sub attributes.
+
+### 1.4. Event Formats
+
+Event formats, used with the _structured_ content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][json-format], but MAY support
+any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for Pub/Sub, or
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+This mapping is intentionally robust against changes, including the addition and
+removal of event attributes, and also accommodates vendor extensions to the
+event metadata.
+
+### 2.1. datacontenttype Attribute
+
+The `datacontenttype` attribute is assumed to contain a [RFC2046][rfc2046]
```

Comment:
"is assumed to" doesn't have a clear meaning. Can we replace that phrase and others with RFC2119 words?

- "is free to"
- "the assumption is"
- "the expectation is"

Reply:
Makes sense, other specs used this terminology, but let's make it clearer here..